### PR TITLE
bugfix: when mixed use of AT and TCC, AT branchs is not deleted

### LINKED ISSUE #3295: AT和TCC混用的时候，全局提交的时候，global_table和AT对应的分支事务还有undo_log没删除
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
AT和TCC混用的时候，全局提交的时候，global_table和AT对应的分支事务还有undo_log没删除

### Ⅱ. Describe what happened
关联 #2978 @wangliang181230 
由于AT模式的分支事务没有删除，所有这里的hasBranch就是true了，然后doGlobalCommit返回了false
image
然后success是false，他就进不去这一层进行异步提交了
image
最后就没有执行异步提交删除分支事务和全局事务信息

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by wangliang181230**
### Ⅰ. Describe what this PR did
bugfix: when mixed use of AT and TCC, AT branchs is not deleted.
问题修复：AT和TCC混合使用时，AT分支未被及时删除。

### 顺便优化一下：
顺便添加上一个PR中变更过配置位置的提示，效果如下图：
图片

### Ⅱ. Does this pull request fix one issue?

fixes #3295

### CODE REVIEW

File: server/src/main/java/io/seata/server/coordinator/DefaultCore.java [hunk 1]
```diff
@@ -235,11 +238,12 @@ public boolean doGlobalCommit(GlobalSession globalSession, boolean retrying) thr
                     }
                 }
             }
-            if (globalSession.hasBranch()) {
+            if (hasNoBranchsCanBeCommittedAsync && globalSession.hasBranch()) {
```
Comment (caohdgege) [inline server/src/main/java/io/seata/server/coordinator/DefaultCore.java:241]: 这里用 globalSession.canBeCommittedAsync 会不会好点？

Comment (wangliang181230) [reply server/src/main/java/io/seata/server/coordinator/DefaultCore.java]: done

**Review by caohdgege — APPROVED** [review]
LGMT

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by jsbxyyx — APPROVED** [review]
LGTM

**Pull request merged**